spec/index.bs

```diff
@@ -1409,62 +1409,63 @@ interface GPUBufferUsage {
+        <pre class=argumentdef for="GPUDevice/createBuffer(descriptor)">
+            |descriptor|:
+        </pre>
 
-    1. Retun an error buffer.
+        **Returns:** {{GPUBuffer}}
 
-    Issue(gpuweb/gpuweb#605): Explain that the resulting error buffer can still be mapped at creation.
+        - If any of the following conditions are unsatisfied, return an error buffer and stop.
+            <div class=validusage>
+                - |this| is a [=valid=] {{GPUDevice}}.
+                - |descriptor|.{{GPUBufferDescriptor/usage}} is a subset of |this|.[[allowed buffer usages]].
+                - If |descriptor|.{{GPUBufferDescriptor/usage}} contains {{GPUBufferUsage/MAP_READ}} then
+                    the only other usage it may contain is {{GPUBufferUsage/COPY_DST}}.
+                - If |descriptor|.{{GPUBufferDescriptor/usage}} contains {{GPUBufferUsage/MAP_WRITE}} then
+                    the only other usage it may contain is {{GPUBufferUsage/COPY_SRC}}.
+                - If |descriptor|.{{GPUBufferDescriptor/mappedAtCreation}} is true then
+                    |descriptor|.{{GPUBufferDescriptor/size}} must be a multiple of 4.
 
-  1. Let |b| be a new {{GPUBuffer}} object.
-  1. Set |b|.{{GPUBuffer/[[size]]}} to |descriptor|.{{GPUBufferDescriptor/size}}.
-  1. Set |b|.{{GPUBuffer/[[usage]]}} to |descriptor|.{{GPUBufferDescriptor/usage}}.
-  1. If |descriptor|.{{GPUBufferDescriptor/mappedAtCreation}} is true:
+                Issue(gpuweb/gpuweb#605): Explain that the resulting error buffer can still be mapped at creation.
 
-    1. Set |b|.{{GPUBuffer/[[mapping]]}} to a new {{ArrayBuffer}} of size |b|.{{GPUBuffer/[[size]]}}.
-    1. Set |b|.{{GPUBuffer/[[mapping_range]]}} to `[0, descriptor.size]`.
-    1. Set |b|.{{GPUBuffer/[[mapped_ranges]]}} to `[]`.
-    1. Set |b|.{{GPUBuffer/[[state]]}} to [=buffer state/mapped at creation=].
+                Issue(gpuweb/gpuweb#605): Explain what are a {{GPUDevice}}'s `[[allowed buffer usages]]`.
+            </div>
 
-  1. Else:
+        - Let |b| be a new {{GPUBuffer}} object.
+        - Set |b|.{{GPUBuffer/[[size]]}} to |descriptor|.{{GPUBufferDescriptor/size}}.
+        - Set |b|.{{GPUBuffer/[[usage]]}} to |descriptor|.{{GPUBufferDescriptor/usage}}.
+        - If |descriptor|.{{GPUBufferDescriptor/mappedAtCreation}} is true:
 
-    1. Set |b|.{{GPUBuffer/[[mapping]]}} to `null`.
-    1. Set |b|.{{GPUBuffer/[[mapping_range]]}} to `null`.
-    1. Set |b|.{{GPUBuffer/[[mapped_ranges]]}} to `null`.
-    1. Set |b|.{{GPUBuffer/[[state]]}} to [=buffer state/unmapped=].
+            - Set |b|.{{GPUBuffer/[[mapping]]}} to a new {{ArrayBuffer}} of size |b|.{{GPUBuffer/[[size]]}}.
+            - Set |b|.{{GPUBuffer/[[mapping_range]]}} to `[0, descriptor.size]`.
+            - Set |b|.{{GPUBuffer/[[mapped_ranges]]}} to `[]`.
+            - Set |b|.{{GPUBuffer/[[state]]}} to [=buffer state/mapped at creation=].
 
-  1. Set each byte of |b|'s allocation to zero.
-  1. Return |b|.
+        - Else:
```

Comment:
Unrelated, but I think we should make the "if" and the "else" be in the same bullet. I think this would work:
```suggestion
            Else:
```

Reply:
Yup! That works.